assertj-core/src/test/java/org/assertj/core/error/ShouldBeEqual_Test.java

```diff
@@ -66,6 +66,18 @@ void should_use_actual_and_expected_representation_in_AssertionFailedError_actua
                                   " but was: [1, 2, 3]"));
   }
 
+  @Test
+  void should_display_class_for_ambiguous_CharSequence() {
```

Comment:
Not completely sure where the 'right' place for this test is. All these test classes seem to perform variants of ambiguous `toString` checks:
- `ShouldBeEqual_Test` (this class here)
- `ShouldBeEqual_newAssertionError_differentiating_expected_and_actual_Test`
- `StandardRepresentation_unambiguousToStringOf_Test`

Reply:
That would be in `StandardRepresentation_unambiguousToStringOf_Test` where we should have exhaustive tests for representation, the other tests are just to check that `ShouldBeEqual` honors that too but we only one test proving that.